x-pack/plugins/gis/public/shared/layers/sources/ems_tms_source/ems_tms_source_editor.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import React from 'react';
+import {
+  EuiSelect,
+  EuiFormRow,
+} from '@elastic/eui';
+
+import { getEmsTMSServices } from '../../../../meta';
+
+
+export class EMSTMSCreateSourceEditor extends React.Component {
+
+
+  state = {
+    emsTmsOptionsRaw: null
+  };
+
+  _loadTmsOptions = async () => {
+    const options = await getEmsTMSServices();
+    if (this._isMounted) {
+      this.setState({
+        emsTmsOptionsRaw: options
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+    this._loadTmsOptions();
+  }
+
+  render() {
+
+    if (!this.state.emsTmsOptionsRaw) {
+      return null;
```

Comment:
Same as above, lets display a loading message so users know why the editor is blank